### ISSUE TITLE
BGL PR bounty hunt ($10000 overall budget)

### ISSUE THREAD

**Issue by wu-emma** [BitgesellOfficial/bitgesell#39]
To get more people involved and provide motivation, we are announcing Bitgesell Pull Request bounty hunt!

The rules are simple:

- You can create any reasonable pull request that may contain any modifications, including, but not limited to:
  - Refactoring and simplification;
  - Test fixes (1 test group/file fixed by single PR counts!);
  - Cleanup of features that are no longer used (e.g. non-segwit transactions);
  - Documentation and comments (but if no code changes then some reasonable amount of changes should be done to count);
  - Bug fixes;
- Every PR would be paid $500 in USDT upon approval, larger PRs could receive bigger payouts (PRs of lesser significance like comments editing or one-line changes could receive lesser payout).
- Interested people are welcome to become maintainers of the project;

**Comment by ayushkumar63123** [comment]
Please tell whether my pr refactoring extract_strings.py would be rewarded or not?

**Comment by ayushkumar63123** [comment]
I am not getting any reply from one week or more. Please reply

**Comment by ayushkumar63123** [comment]
0x6957d999Bbea7771C22beA4Dea4d7c0E39E8Aa1F
Please note my wallet here too (I also sent it on mail).

**Comment by ayushkumar63123** [comment]
I made a new pr please check it out

**Comment by berbagi12** [comment]
if it is considered not in use the feature can be deleted and needs cleaning.  
something that has no function can be deleted.

**Comment by ayushkumar63123** [comment]
There was an issue stating it needed to be deleted so I did if I have not
created any further issues and have solved the bug, then can it be
considered for reward?

**Comment by ayushkumar63123** [comment]
Is this pr eligible for reward?

**Comment by ayushkumar63123** [comment]
Please reply whether this is eligible or not then I will continue resolving other issues too

**Comment by ayushkumar63123** [comment]
By eligible I mean for bug bounty of 500 usdt

**Comment by ayushkumar63123** [comment]
Please review my pull request.

**Comment by antonyip** [comment]
Hi, Would like to check if this is normal, or is something wrong with my setup? 
![image](https://user-images.githubusercontent.com/9086733/126585723-638fdfdb-a5a6-4590-bbc7-8206f01a2a85.png)

**Comment by Developer-piyush** [comment]
> To get more people involved and provide motivation, we are announcing Bitgesell Pull Request bounty hunt!
> 
> The rules are simple:
> 
> * You can create any reasonable pull request that may contain any modifications, including, but not limited to:
>   
>   * Refactoring and simplification;
>   * Test fixes (1 test group/file fixed by single PR counts!);
>   * Cleanup of features that are no longer used (e.g. non-segwit transactions);
>   * Documentation and comments (but if no code changes then some reasonable amount of changes should be done to count);
>   * Bug fixes;
> * Every PR would be paid $500 in USDT upon approval, larger PRs could receive bigger payouts (PRs of lesser significance like comments editing or one-line changes could receive lesser payout).
> * Interested people are welcome to become maintainers of the project;

**Sir please check my PR #47 **

**Comment by Developer-piyush** [comment]
> To get more people involved and provide motivation, we are announcing Bitgesell Pull Request bounty hunt!
> 
> The rules are simple:
> 
> * You can create any reasonable pull request that may contain any modifications, including, but not limited to:
>   
>   * Refactoring and simplification;
>   * Test fixes (1 test group/file fixed by single PR counts!);
>   * Cleanup of features that are no longer used (e.g. non-segwit transactions);
>   * Documentation and comments (but if no code changes then some reasonable amount of changes should be done to count);
>   * Bug fixes;
> * Every PR would be paid $500 in USDT upon approval, larger PRs could receive bigger payouts (PRs of lesser significance like comments editing or one-line changes could receive lesser payout).
> * Interested people are welcome to become maintainers of the project;

sir please review my pr #47

**Comment by Developer-piyush** [comment]
> To get more people involved and provide motivation, we are announcing Bitgesell Pull Request bounty hunt!
> 
> The rules are simple:
> 
> * You can create any reasonable pull request that may contain any modifications, including, but not limited to:
>   
>   * Refactoring and simplification;
>   * Test fixes (1 test group/file fixed by single PR counts!);
>   * Cleanup of features that are no longer used (e.g. non-segwit transactions);
>   * Documentation and comments (but if no code changes then some reasonable amount of changes should be done to count);
>   * Bug fixes;
> * Every PR would be paid $500 in USDT upon approval, larger PRs could receive bigger payouts (PRs of lesser significance like comments editing or one-line changes could receive lesser payout).
> * Interested people are welcome to become maintainers of the project;

I request you to please have a look at my PR🙏🙏

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__Work for 500.0 USDT (500.00 USD @ $1.0/USDT) has been submitted by__:

 1. [@setiaj700](https://gitcoin.co/setiaj700)
 2. [@ayushkumar63123](https://gitcoin.co/ayushkumar63123)
 3. [@berbagi12](https://gitcoin.co/berbagi12)
 4. [@antonyip](https://gitcoin.co/antonyip)
 5. [@developer-piyush](https://gitcoin.co/developer-piyush)
 6. [@soptq](https://gitcoin.co/soptq)



@wu-emma please take a look at the submitted work:
* [PR](https://github.com/BitgesellOfficial/bitgesell/pull/51) by @soptq
* [PR](https://github.com/BitgesellOfficial/bitgesell/pull/47) by @developer-piyush
* [PR](https://github.com/BitgesellOfficial/bitgesell/pull/46) by @antonyip
* [PR](https://github.com/BitgesellOfficial/bitgesell/pull/44) by @berbagi12
* [PR](https://github.com/BitgesellOfficial/bitgesell/pull/44) by @ayushkumar63123
* [PR](https://gitcoin.co/setiaj700) by @setiaj700

<hr>

* Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/BitgesellOfficial/bitgesell/39/100026119)
* Want to chip in? Add your own contribution [here](https://gitcoin.co/issue/BitgesellOfficial/bitgesell/39/100026119).
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://discord.gg/gitcoin/'>Gitcoin's Discord</a>
* $3,459,871.36 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__Work has been started__.


These users each claimed they can complete the work by 265 years, 3 months from now.
Please review their action plans below:


**1) [ranjiths10](https://gitcoin.co/ranjiths10) has started work.**

Submit valuable and easy-to-understand bugs. 
Earn and show respect. 
Get more comfortable with the basics. 
Write small, vulnerable programs and challenge others to find the hidden vulnerabilities.
**2) [ayushkumar63123](https://gitcoin.co/ayushkumar63123) has started work.**

I removed support for p2pk transaction in wallet.py and also the ability to sign for p2pk since its a non-segwit transaction type.
**3) [tafadzwachikuvire](https://gitcoin.co/tafadzwachikuvire) has started work.**

i will make it work by coding .look i am a boy aged 14 and i was born on 2 may 2007 so i will be doing this task .it may take long but start to expect this done is 4 months thank you
**4) [gconnect](https://gitcoin.co/gconnect) has started work.**

I would like to create a contribution guide for this project.
**5) [antonyip](https://gitcoin.co/antonyip) has started work.**

Update and fix p2p_* tests......
**6) [developer-piyush](https://gitcoin.co/developer-piyush) has started work.**

I have updated readme because first impression is last impression for users/funders/developers and visitors!
**7) [soptq](https://gitcoin.co/soptq) has started work.**

I can set up a build test to improve the general quality of later PR.
**8) [slowriot](https://gitcoin.co/slowriot) has started work.**

Miscellaneous code quality and documentation improvements

Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/BitgesellOfficial/bitgesell/39/100026119).

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__The funding of 500.0 USDT (500.00 USD @ $1.0/USDT)  attached to this issue has been approved & issued to @slowriot.__  

 
* Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/bitgesellofficial/bitgesell/39/100026119)
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://discord.gg/gitcoin/'>Gitcoin's Discord</a>
* $6,951,403.67 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__The funding of 500.0 USDT (500.00 USD @ $1.0/USDT)  attached to this issue has been approved & issued to @slowriot.__  

 
* Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/bitgesellofficial/bitgesell/39/100026119)
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://discord.gg/gitcoin/'>Gitcoin's Discord</a>
* $6,951,403.67 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by slowriot** [comment]
> Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done

Please note I still have the following merge requests open at present, all of which have been [open for several months](https://github.com/BitgesellOfficial/bitgesell/issues/32#issuecomment-913701191):

- https://github.com/BitgesellOfficial/bitgesell/pull/52 - paid ✔️
- https://github.com/BitgesellOfficial/bitgesell/pull/53 - awaiting proper review
- https://github.com/BitgesellOfficial/bitgesell/pull/54 - awaiting proper review
- https://github.com/BitgesellOfficial/bitgesell/pull/55 - awaiting proper review
- https://github.com/BitgesellOfficial/bitgesell/pull/57 - awaiting proper review
- https://github.com/BitgesellOfficial/bitgesell/pull/58 - awaiting proper review
- https://github.com/BitgesellOfficial/bitgesell/pull/59 - awaiting proper review

Additionally, I have a PR to join some of the above changes into Janus branch, where development appears to be ahead of master:
- https://github.com/BitgesellOfficial/bitgesell/pull/56

I have further contributions already committed locally, and am eager to submit them, but have not yet done so pending resolution of the discussion at https://github.com/BitgesellOfficial/bitgesell/pull/52#issuecomment-911791689

Considering the amount of time that has passed, and the low review effort from maintainers, I have little faith that this project is serious about paying its contributors. I would caution other developers to consider carefully before contributing for the bounty.

**Comment by ayushkumar63123** [comment]
Someone please check my bounty

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__The funding of 500.0 USDT (500.00 USD @ $1.0/USDT)  attached to this issue has been approved & issued to @slowriot.__  

 
* Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/26119)
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://discord.gg/gitcoin/'>Gitcoin's Discord</a>
* $3,312,497.77 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__The funding of 500.0 USDT (500.00 USD @ $1.0/USDT)  attached to this issue has been approved & issued to @slowriot.__  

 
* Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/26119)
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://discord.gg/gitcoin/'>Gitcoin's Discord</a>
* $3,311,997.77 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__The funding of 500.0 USDT (500.00 USD @ $1.0/USDT)  attached to this issue has been approved & issued to @slowriot.__  

 
* Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/26119)
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://discord.gg/gitcoin/'>Gitcoin's Discord</a>
* $3,311,997.77 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by slowriot** [comment]
Contrary to the above three notifications, I have not received any further payment.  PRs remain open.

Contributors beware.

**Comment by wu-emma** [comment]
@madnadyka @janus @van-orton 
Please take care of this. @slowriot thanks for your patience and understanding. We would sort this out.

**Comment by slowriot** [comment]
> @madnadyka @janus @van-orton Please take care of this. @slowriot thanks for your patience and understanding. We would sort this out.

I see that after almost a year, the following two pull requests have finally been merged:

- https://github.com/BitgesellOfficial/bitgesell/pull/52
- https://github.com/BitgesellOfficial/bitgesell/pull/53

I'm glad to see my contributions are being used.  However, I have not yet received payment.  I would appreciate it if you could process payment to the addresses given in the PRs as agreed under this bounty.

## Contributors:

For the benefit of other developers considering contributing, and wondering about the legitimacy of this project - I will update this message if payment has been received.

**Comment by wu-emma** [comment]
> > @madnadyka @janus @van-orton Please take care of this. @slowriot thanks for your patience and understanding. We would sort this out.
> 
> I see that after almost a year, the following two pull requests have finally been merged:
> 
>     * [URL fixes, documentation improvements #52](https://github.com/BitgesellOfficial/bitgesell/pull/52)
> 
>     * [Base58 fixes #53](https://github.com/BitgesellOfficial/bitgesell/pull/53)
> 
> 
> I'm glad to see my contributions are being used. However, I have not yet received payment. I would appreciate it if you could process payment to the addresses given in the PRs as agreed under this bounty.
> ## Contributors:
> 
> For the benefit of other developers considering contributing, and wondering about the legitimacy of this project - I will update this message if payment has been received.

We appreciate the effort done and thanks for participation. However, having considered the PRs opened, after reviewing and extracting what could be merged into main branch, a decision was made to provide compensation in the amount of $300. Sorry for delayed reviews.

**Comment by JeremyZeng77** [comment]
Submitted two documentation cleanup PRs that should fit this bounty's documentation / cleanup category:

| PR | Scope | Verification |
| --- | --- | --- |
| https://github.com/BitgesellOfficial/bitgesell/pull/265 | Fix missing README translation link and point active translation docs at the BGL Transifex project/source file | `git diff --check`; checked missing `README-zh.md`; checked `BGL_en.ts`; raw GitHub URL returned HTTP 200 |
| https://github.com/BitgesellOfficial/bitgesell/pull/266 | Retarget `CONTRIBUTING.md` from libsecp256k1 guidance to Bitgesell Core contributor guidance | `git diff --check`; checked stale wording; checked local markdown links |

Payment if approved: USDT TRC20 `TWupTAoFjxR2VmqUUvzhfmUwXVZR6Kvz1Y`.

**Comment by JeremyZeng77** [comment]
Submitted another non-overlapping README cleanup for this bounty:

- PR: https://github.com/BitgesellOfficial/bitgesell/pull/267
- Scope: replaces a stale libsecp256k1 examples/benchmark section with existing Bitgesell Core developer documentation links.
- Verification: `git diff --check`; confirmed `examples` is absent; confirmed target docs exist; confirmed stale `examples/*` and libsecp256k1 benchmark references are removed from `README.md`.

Payment if approved: USDT TRC20 `TWupTAoFjxR2VmqUUvzhfmUwXVZR6Kvz1Y`.

**Comment by Asobu01** [comment]
Happy to work on BitgesellOfficial/bitgesell#39.
For "BGL PR bounty hunt ($10000 overall budget)", I will start with reproducing the reported issue first, then applying a focused fix with clear validation.
I can share a quick checkpoint first, then deliver a draft PR in ~8h.
Planned deliverables: repro note with concrete evidence (logs/trace when relevant); focused patch (plus tests/docs if needed); review-ready PR with validation notes.
I keep changes minimal and aligned with existing style.
I can post a reproducible checklist/log snippet in ~30 minutes so you can verify execution speed before the full PR.
If this direction matches your expectation, I will start immediately. @Asobu01

**Comment by iamdinhthuan** [comment]
Submitted PR for this bounty: https://github.com/BitgesellOfficial/bitgesell/pull/268

Scope: adds a focused `deploymentinfo_tests` unit test that asserts each `VersionBitsDeploymentInfo` entry has a non-null, non-empty name and that `DeploymentName()` matches the metadata table. This is regression coverage for the missing deployment-name failure class discussed in #123.

Verification: red check by temporarily setting `taproot_discarded` to `nullptr` failed with `missing deployment name at index 1`; then `make -j2 -C src test/test_BGL`; `LD_LIBRARY_PATH=/home/thuan/miniconda3/envs/py311/lib src/test/test_BGL --run_test=deploymentinfo_tests` returned `No errors detected`; `git diff --cached --check` passed.

Payout if approved: USDT or USDC SPL on Solana `6mL4QQwcwLmb9UX1PjHtRSQ8J3HazCeVSDdZdDVQGXJj`.

**Comment by Asobu01** [comment]
Quick follow-up on `BitgesellOfficial/bitgesell#39`. I am still ready to start immediately and can post the first repro/checklist note before the draft PR if helpful. If you want me on this one, I can begin now. (@Asobu01)

**Comment by Sikkra** [comment]
Submitted PR #270 for the PR bounty: https://github.com/BitgesellOfficial/bitgesell/pull/270

Summary: makes `test/functional/test_runner.py --help` work in a fresh checkout before `test/config.ini` exists, reports a clean missing-config error for non-help runs, and avoids creating temporary test directories for help-only exits.

Verification run locally:
- `python -m py_compile test\functional\test_runner.py`
- `python test\functional\test_runner.py --help` exits 0 before configure
- `python test\functional\test_runner.py --jobs 1` exits 2 with a clear missing-config message before configure

USDT/USDC-compatible EVM payout address: `0xa925FdD65a0f34bb415Bae1c57536Be33AbCfA92`

**Comment by Sikkra** [comment]
Submitted another PR for the PR bounty: https://github.com/BitgesellOfficial/bitgesell/pull/271

Summary: finishes the Bitgesell utility binary rename in the util test runner and test vectors. The util test data had a remaining block of `./bitcoin-tx` entries, and the runner only supported `BITCOINUTIL` / `BITCOINTX` overrides. The PR switches those paths to `./BGL-tx`, `BGLUTIL`, and `BGLTX`.

Verification run locally:
- `python -m py_compile test\util\test_runner.py`
- parsed `test/util/data/BGL-util-test.json` and confirmed all 103 test entries now use only `./BGL-util` or `./BGL-tx`

USDT/USDC-compatible EVM payout address: `0xa925FdD65a0f34bb415Bae1c57536Be33AbCfA92`

**Comment by JeremyZeng77** [comment]
Submitted PR for the bounty hunt: https://github.com/BitgesellOfficial/bitgesell/pull/272

Payout preference: USDT (TRC20) TWupTAoFjxR2VmqUUvzhfmUwXVZR6Kvz1Y.

**Comment by RYB-404** [comment]
Submitted PR #273 for this bounty: https://github.com/BitgesellOfficial/bitgesell/pull/273

It makes the include-guard lint pass on the current tree by excluding imported/generated headers and fixing BGL-owned guard mismatches. The PR body includes /claim #39 and verification details.

**Comment by tolga-tom-nook** [comment]
Submitted another focused Bitgesell improvement PR for the bounty program:

PR: https://github.com/BitgesellOfficial/go-bgld/pull/2

Scope: fixes `go-bgld` `GetBlockheader` decoding for Core-shaped JSON responses. Bitgesell Core returns `bits` as a compact hex string like `"1d00ffff"`, and the previous/next block hash fields use lower-case JSON keys. The PR updates the struct decoding and adds an httptest regression for that response shape.

Validation:
- RED proof on upstream with the new regression: `go test ./...` fails before the fix because `BlockHeader.Bits` is `uint32` while Core returns a string.
- GREEN proof after fix: `go test ./...` passes.

Payout address if approved: USDT/EVM-compatible `0x4a76c7E64C08cF29B59eFC640b4ada97A270d428`.

**Comment by tolga-tom-nook** [comment]
Additional bounty/improvement submissions also relevant to this main PR bounty issue, with RED/GREEN proof already posted on #81:

1. https://github.com/BitgesellOfficial/bgld/pull/4
   - Scope: focused Go test/build reliability fix in official `bgld` repo.
   - RED: upstream `go test ./mempool` fails on `t.Fatal` with `%d` vet error.
   - GREEN: PR branch `go test ./mempool` passes.

2. https://github.com/BitgesellOfficial/bitgesell-blockchain-sdk/pull/2
   - Scope: SDK correctness fix in official TypeScript blockchain SDK.
   - RED/proof: upstream `getAddressUTXO()` calls the unconfirmed-transactions endpoint and returns the wrong data shape.
   - GREEN: dedicated endpoint regression test passes; `npx tsc -p tsconfig.json` passes; full Jest suite passes (`6` suites / `23` tests).

3. https://github.com/BitgesellOfficial/gobglbridge/pull/5
   - Scope: WBGL bridge robustness/security fix.
   - RED: malformed but valid hex signature `0x00` panics in `validateMsgSignature` because code reads `sigBytes[64]` before checking decoded length.
   - GREEN: short signatures now return a normal validation error; focused regression test passes; full `go test ./...` passes.

Payout address if approved: USDT/EVM-compatible `0x4a76c7E64C08cF29B59eFC640b4ada97A270d428`.

Assisted by Hermes Agent.

**Comment by tolga-tom-nook** [comment]
Submitting another PR for the Bitgesell improvement bounty:

PR: https://github.com/BitgesellOfficial/bglapiserver/pull/1
Repo: `BitgesellOfficial/bglapiserver`

Fix: transaction API routes were accidentally nested under the `blocks_data` feature flag in `api/routes.py`, so enabling `transaction` without `blocks_data` skipped the core transaction lookup/proof endpoints. This PR dedents those registrations so the `transaction` option controls them independently.

Validation:

```text
PYTHONDONTWRITEBYTECODE=1 python3 -m pytest tests/test_route_registration.py -q
1 passed in 0.02s

Python AST syntax OK
git diff --check: clean
```

Payout address for USDT/EVM-compatible rails: 0x4a76c7E64C08cF29B59eFC640b4ada97A270d428

**Comment by tolga-tom-nook** [comment]
Submitted a wbgl-bridge-sdk fix under this bounty/improvement program:

- PR: https://github.com/BitgesellOfficial/wbgl-bridge-sdk/pull/2
- Scope: fixes BGL -> WBGL bridge submit endpoint construction. The SDK default/config example uses `https://bglswap.com/app/`; the previous code appended `/submit/bgl`, producing `https://bglswap.com/app//submit/bgl`. The PR normalizes trailing slashes so both trailing- and non-trailing-slash bridge endpoints resolve to canonical `/app/submit/bgl`.
- Validation: focused Jest regression added and passing (`2 passed`), targeted ESLint passing, `git diff --check` clean.
- Note: full `npm run build` is currently blocked by a repo dependency/toolchain mismatch (`@types/node` disposable symbols vs pinned TypeScript 4.9), unrelated to this change.

Payout address if approved: EVM/BSC USDC `0x4a76c7E64C08cF29B59eFC640b4ada97A270d428`.

**Comment by tolga-tom-nook** [comment]
Submitted a bitgesell-blockchain-sdk fix under this bounty/improvement program:

- PR: https://github.com/BitgesellOfficial/bitgesell-blockchain-sdk/pull/3
- Scope: fixes `getMempoolState()` to call the aggregate `/mempool/state` endpoint instead of the paginated `/mempool/transactions` endpoint, and strengthens the regression test so it awaits the SDK call and asserts state fields (`inputs`, `outputs`, `transactions`).
- RED proof: strengthened focused Jest test failed before the code fix because the method returned transaction-list fields (`count`, `page`, `list`) instead of state fields.
- GREEN proof: focused mempool-state test passed, full `mempool.test.ts` passed, full Jest suite passed (`5` suites / `22` tests), `npm run build` passed, and `git diff --check` is clean.
- Note: targeted ESLint currently reports pre-existing repo config/rule issues unrelated to this change (`functional/no-class` in the existing class-based file and test parser-project inclusion).

Payout address if approved: EVM/BSC USDC `0x4a76c7E64C08cF29B59eFC640b4ada97A270d428`.

**Comment by tolga-tom-nook** [comment]
Claim/proof for Bitgesell improvement bounty:

PR: https://github.com/BitgesellOfficial/jsbgl/pull/4

Scope: fixes `jsbgl` Bech32/SegWit address validation so mixed-case addresses are rejected before checksum normalization, while all-lowercase and all-uppercase Bech32 remain valid per BIP173. The previous implementation lowercased the HRP/payload for checksum validation and could accept mixed-case BGL Bech32 strings.

Validation evidence:
- RED: added regression made `npm test` fail in `Address functions: isAddressValid` with `AssertionError: expected true to equal false` for mixed-case `Bgl1QW508D6QEJXTDG4Y5R3ZARVARY0C5XW7K0FY5A3`.
- GREEN: `npm test` passes (`68 passing`).
- Package/dist proof: package main `require('./')` rejects the mixed-case address and accepts lowercase/uppercase forms.
- `git diff --check` clean.

Payout address if approved and USDT/EVM-compatible payout is used: `0x4a76c7E64C08cF29B59eFC640b4ada97A270d428`

**Comment by MylloVinyllo** [comment]
Hi, I’m considering a focused PR under this bounty.

Before I spend implementation time, can you confirm whether this PR bounty is still active for new contributions in May 2026, and whether small focused test/CI/cleanup PRs are still eligible for review and payment if accepted?

If confirmed, I’ll keep the change narrow, include clear validation, and submit a focused PR rather than a broad rewrite.

**Comment by TUPM96** [comment]
Submitted a focused Bitgesell improvement PR under this bounty program:

- PR: https://github.com/BitgesellOfficial/js-bitgesellcore-rpc/pull/2
- Scope: fixes the default debug logger namespace in the JS RPC client from `bitcoin-core` to `bitgesell-core`, matching the package/docs and making documented `DEBUG=bitgesell-core` logging work for default clients.
- Validation: `npx mocha test/default_logger_test.js test/logging/request-obfuscator_test.js --require should`; `npx eslint src/index.js test/default_logger_test.js`; `git diff --check`.

Note: regular `npm install` currently triggers the existing `dependencies` lifecycle script and fails without docker-compose config; I used `npm install --ignore-scripts` for local verification and left that separate issue to PR #1.

Payout details can be provided privately if approved.

**Comment by tolga-tom-nook** [comment]
Submitted another focused Bitgesell improvement PR for this program:

- PR: https://github.com/BitgesellOfficial/js-bitgesellcore-rpc/pull/3
- Scope: restores inherited `config.bitcoin*` aliases to the Bitgesell service configs so legacy tests/consumers reference the same Bitgesell objects; adds a no-daemon regression test and normalizes package-lock root metadata.
- Validation: `./node_modules/.bin/eslint src test --quiet`, `npm test -- --grep 'Configuration aliases'`, and `git diff --check` all passed.
- Payout address if approved: EVM/USDT-compatible `0x4a76c7E64C08cF29B59eFC640b4ada97A270d428`.

**Comment by tolga-tom-nook** [comment]
Submitted PR for this bounty/improvement lane:

- PR: https://github.com/BitgesellOfficial/bglutil/pull/4
- Scope: restores the missing `github.com/bitgesellofficial/bgld` module dependency and refreshed sums so a clean `bglutil` checkout can resolve Bitgesell-specific `bgld/...` imports instead of failing before tests start.
- Validation: `GOTOOLCHAIN=local go test ./base58 ./bech32 ./gcs ./gcs/builder ./hdkeychain` passed; `git diff --check HEAD~1..HEAD` clean.
- Payout address if approved: `0x4a76c7E64C08cF29B59eFC640b4ada97A270d428`

**Comment by manuelsampedro1** [comment]
Submitted a focused Rust RPC client fix under this bounty program:

- PR: https://github.com/BitgesellOfficial/rust-bitgesellcore-rpc/pull/2
- Scope: fixes clean workspace builds by aligning the client dependency/imports with the local `bitgesellcore-rpc-json` crate name instead of the stale `bitcoincore-rpc-json` name.
- Validation: `cargo test --workspace --lib`, `cargo test --workspace`, and `cargo test --workspace --examples`.

**Comment by manuelsampedro1** [comment]
Submitted another focused Bitgesell improvement PR under this bounty program:

- PR: https://github.com/BitgesellOfficial/brotlic/pull/1
- Scope: improves the git-checkout build failure for brotlic by detecting a missing Brotli C submodule before cc/clang fails, and documents `git submodule update --init --recursive`.
- Validation: `git submodule update --init --recursive`; `cargo test`.

**Comment by tolga-tom-nook** [comment]
Submitted PR for bounty consideration: https://github.com/BitgesellOfficial/bitgesell-arm/pull/1

Scope: fixes `docker-entrypoint.sh` so option and `.conf` invocations are rewritten to the repository's actual `bgl_oneshot` command instead of `bgl`, preventing container startup failure for standard option/config-file forms. Adds focused regression coverage for both forms.

Verification: `bash test/tests/entrypoint.sh` passed; `git diff --check origin/main..HEAD` clean.

Payout address if approved: `0x4a76c7E64C08cF29B59eFC640b4ada97A270d428`

**Comment by 1507819106zxzx-crypto** [comment]
Submitted a focused jsbgl validation fix: https://github.com/BitgesellOfficial/jsbgl/pull/5

It makes addressToHash reject checksum-invalid P2PKH, P2SH, and Bech32 addresses before decoding, and keeps addressToScript from building scripts from invalid address hashes. Regression tests were added for the invalid checksum cases.

Verification: npm test and npm run build passed locally.

Payout address if approved: 0x4451dF3D21925eF7a62D20eEdc80B99f7140C5D2

**Comment by 1507819106zxzx-crypto** [comment]
Submitted another focused Bitgesell library hardening PR for bounty consideration: https://github.com/BitgesellOfficial/bitgesell-bip39/pull/2

Scope: rejects malformed entropy hex strings in bitgesell-bip39 before Node's Buffer.from(..., 'hex') can silently truncate odd-length strings or stop at non-hex characters. This prevents malformed input from producing a mnemonic for different entropy than the caller supplied.

Regression coverage: odd-length, non-hex, and empty string entropy now throw the existing Invalid entropy error.

Verification:
- npx tsc -p tsconfig.json
- npx prettier ts_src/index.ts --check
- npm run lint
- npm run unit (386 passing)
- git diff --check

Payout address if approved: 0x4451dF3D21925eF7a62D20eEdc80B99f7140C5D2

**Comment by 1507819106zxzx-crypto** [comment]
Submitted another focused maintenance PR for the bounty program: https://github.com/BitgesellOfficial/bitgesell-blockchain-sdk/pull/4

Change: removes a committed describe.only from the SDK address test suite so the full Jest suite runs again, including blockchain, mempool, transaction, API, and address tests.

Verification:
- npm run build:main
- npm test -- --runInBand (5 suites, 22 tests passing)
- git diff --check HEAD~1..HEAD

Payout address if accepted: 0x4451dF3D21925eF7a62D20eEdc80B99f7140C5D2

**Comment by 1507819106zxzx-crypto** [comment]
Submitted another bridge SDK maintenance PR for the bounty program: https://github.com/BitgesellOfficial/wbgl-bridge-sdk/pull/3

Change: removes committed focused Jest tests, mocks default WBGLBridgeSDK status/balance/contract reads, and gates real BGL/WBGL swap integration tests behind explicit env flags so normal CI does not require private keys or spend gas.

Verification:
- npm test -- --runInBand (1 suite passed, 2 integration suites skipped by default; 8 passed, 4 skipped)
- npx tsc -p tsconfig.json --skipLibCheck
- git diff --check

Payout address if accepted: 0x4451dF3D21925eF7a62D20eEdc80B99f7140C5D2

**Comment by 1507819106zxzx-crypto** [comment]
Submitted another focused wallet hardening PR for bounty consideration: https://github.com/BitgesellOfficial/bitgesell-wallet/pull/5

Scope: escapes transaction IDs and wallet addresses before rendering them into table HTML attributes/text, and URL-encodes explorer/hash-route path segments generated from those table values. Normal BGL addresses and tx hashes render unchanged, while abnormal local-storage/API values cannot break attributes or inject markup into the wallet tables.

Verification:
- node --check src/utils.js
- node --check src/transactions/transactions.js
- node --check src/my-addresses/my-addresses.js
- npx eslint --rule 'linebreak-style: 0' src/utils.js src/transactions/transactions.js src/my-addresses/my-addresses.js
- git diff --check

Payout address if accepted: 0x4451dF3D21925eF7a62D20eEdc80B99f7140C5D2

**Comment by manuelsampedro1** [comment]
Submitted another focused Bitgesell improvement PR under this bounty program:

- PR: https://github.com/BitgesellOfficial/bitgesell-wallet/pull/6
- Scope: hardens the wallet build script by failing early when the dist folder is omitted and quoting dist-folder path expansions so output paths with spaces do not break the build.
- Validation: `bash -n build.sh`; missing-argument usage check; stubbed build-tool smoke test against `/tmp/bitgesell wallet dist test`.

**Comment by tolga-tom-nook** [comment]
Submitted another non-overlapping Bitgesell ecosystem fix for the improvement/PR bounty program:

- PR: https://github.com/BitgesellOfficial/bitgesell-utxo-dump/pull/2
- Scope: restores a clean Go module build for `bitgesell-utxo-dump` by adding `go.mod`/`go.sum`, updating stale module imports to `github.com/BitgesellOfficial/bitgesell-utxo-dump`, and fixing a formatting misuse caught by `go test`.
- RED upstream proof: `go test ./...` fails on `origin/main` with `pattern ./...: directory prefix . does not contain main module or its selected dependencies`.
- GREEN proof: `go test ./...` passes on the PR branch across all packages; `git diff --check origin/main...HEAD` is clean.
- Payout address if accepted: `0x4a76c7E64C08cF29B59eFC640b4ada97A270d428` (EVM/USDT-compatible).

**Comment by tolga-tom-nook** [comment]
Submitted a focused wallet reliability PR for this program:

- PR: https://github.com/BitgesellOfficial/bitgesell-wallet/pull/7
- Scope: `fetchQuery` now runs its `callbackAlways` cleanup when `fetch()` rejects, preventing the send form from staying disabled after network/CORS/RPC failures.
- Validation: `npm test` passed and `git diff --check` is clean.
- Payout address if accepted: `0x4a76c7E64C08cF29B59eFC640b4ada97A270d428` (EVM/USDT-compatible).

**Comment by tolga-tom-nook** [comment]
Submitted a focused bgld-docker entrypoint fix for Bitgesell improvement bounty consideration:

- PR: https://github.com/BitgesellOfficial/bgld-docker/pull/2
- Scope: fixes the stale `btc_oneshot` privilege-drop guard in `docker-entrypoint.sh` so standard option/config startup rewritten to `bgl_oneshot` is re-execed through `gosu BGL` when the container starts as root, rather than running directly as root.
- Validation: RED against upstream failed the new regression with `entrypoint executed bgl_oneshot directly instead of dropping privileges`; GREEN after fix: `sh -n docker-entrypoint.sh`, `sh -n test/entrypoint.sh`, `sh test/entrypoint.sh`, and `git diff --check` all pass.
- Payout address if accepted: EVM/USDT-compatible `0x4a76c7E64C08cF29B59eFC640b4ada97A270d428`.

**Comment by tolga-tom-nook** [comment]
Submitted another focused Bitgesell ecosystem fix for bounty consideration:

- PR: https://github.com/BitgesellOfficial/php-bitgesellcore-rpc/pull/1
- Scope: restores the PHP RPC client's Composer/package identity and autoloadability by aligning `composer.json`, `Config`, tests, and provider class strings with the repo's `naftalimurgor\Bitgesell` namespace; also fixes the `BitgeselldResponse` typo/class-string mismatch and replaces the stale duplicated Go README with PHP usage docs.
- Validation: `python3 -m json.tool composer.json`, static stale-reference scan for old Denpa/Bitcoin and typo symbols, and `git diff --check` all passed. PHP/Composer were unavailable in this runner, so PHPUnit could not be executed here.
- Payout address if accepted: `0x4a76c7E64C08cF29B59eFC640b4ada97A270d428` (EVM/USDT-compatible).

**Comment by manuelsampedro1** [comment]
Submitted another focused Bitgesell improvement PR under this bounty program:

- PR: https://github.com/BitgesellOfficial/bglapiserver/pull/2
- Scope: hardens bglapi Docker run templates by quoting computed host-volume paths, so checkouts under directories with spaces are passed to Docker as single `-v` arguments.
- Validation: `/bin/sh -n` on the touched templates; `python3 -m compileall -q api app node test`; stubbed Docker smoke test from `/tmp/bgl api script test/{api,app,postgres}`.

**Comment by newmattock** [comment]
Submitted a focused pybgl Bech32 validation fix under the Bitgesell PR bounty lane:

- PR: https://github.com/BitgesellOfficial/pybgl/pull/4
- Scope: enforces Bech32/SegWit address case rules by accepting valid all-uppercase addresses, rejecting mixed-case addresses before checksum normalization, and allowing digits in uppercase payloads.
- Validation: reproduced that a generated lowercase bgl1 address was valid while an uppercase variant was incorrectly rejected and a mixed-case payload variant was incorrectly accepted before the fix; added active regression tests. python3 -m unittest pybgl.test.bech32_address_validation, python3 tests.py, and git diff --check all pass.

Payment/review notes: payout details can be provided if approved or requested.

**Comment by tolga-tom-nook** [comment]
Submitted a focused pybgl repository-link cleanup PR for this bounty program:

- PR: https://github.com/BitgesellOfficial/pybgl/pull/5
- Scope: updates README/install docs/Sphinx metadata/package URL from stale `bitaps-com/pybgl` links to the canonical `BitgesellOfficial/pybgl` repo, including the issue tracker.
- Validation: `python3 -m compileall -q pybgl`; stale-link scan for `bitaps-com/pybgl` in README/setup/docs returned no matches; `git diff --check` passed.
- Payout address if accepted: `0x4a76c7E64C08cF29B59eFC640b4ada97A270d428` (EVM/USDT-compatible).

**Comment by manuelsampedro1** [comment]
Submitted another focused Bitgesell improvement PR under this bounty program:

- PR: https://github.com/BitgesellOfficial/go-bitgesell-toolkit/pull/2
- Scope: fixes `GetAddressUTXO` so it calls `/address/utxo/{address}` instead of the unconfirmed-transactions endpoint, with the unit-test mock path updated to cover the regression.
- Validation: static endpoint cross-check with `rg` across go-bitgesell-toolkit, bitgesell-wallet, and bglapiserver route usage. Go tests were not run locally because this environment does not have the Go toolchain installed.

**Comment by manuelsampedro1** [comment]
Submitted another focused Bitgesell improvement PR under this bounty program:

- PR: https://github.com/BitgesellOfficial/php-bitgesellcore-rpc/pull/2
- Scope: replaces the stale duplicated Go `bgld` README with PHP-specific setup, usage, async, configuration, and PHPUnit documentation for the Bitgesell Core RPC client.
- Validation: checked for remaining stale Go/bitcoind references with `rg`; `git diff --check`. Docs-only change, so no PHP suite was run locally.

**Comment by tolga-tom-nook** [comment]
Submitted another focused wallet safety/reliability PR for Bitgesell improvement bounty consideration:

- PR: https://github.com/BitgesellOfficial/bitgesell-wallet/pull/8
- Scope: hardens the send form preflight so submit-time validation rechecks the recipient address, recalculates spend/new-balance state, and refuses to call `sendrawtransaction` when no serialized transaction has been generated. This prevents null/stale transaction RPC submissions if the form is submitted before the earlier field-change validation path runs.
- Validation: `node --check src/send/send.js`, `npm test`, and `git diff --check` all passed.
- Payout address if accepted: `0x4a76c7E64C08cF29B59eFC640b4ada97A270d428` (EVM/USDT-compatible).

**Comment by tolga-tom-nook** [comment]
Submitted another focused bglapiserver README cleanup for Bitgesell bounty consideration:

- PR: https://github.com/BitgesellOfficial/bglapiserver/pull/3
- Scope: refreshes stale README links from `bitaps-com/pybgl` and `bitaps-com/bglapiserver` to the canonical Bitgesell repositories, replaces Bitcoin-specific API module wording with Bitgesell-specific wording, and fixes nearby README grammar.
- Validation: `python3` stale-link/wording scan passed; `git diff --check` passed.
- Payout address if accepted: `0x4a76c7E64C08cF29B59eFC640b4ada97A270d428` (EVM/USDT-compatible).

Assisted by Hermes Agent.

**Comment by CharlieLZ** [comment]
Submitted a focused js-bitgesellcore-rpc reliability fix for bounty consideration:

- PR: https://github.com/BitgesellOfficial/js-bitgesellcore-rpc/pull/4
- Scope: response logging no longer throws when a JSON-RPC batch response contains an unmatched id; matched sensitive results are still obfuscated and unmatched response entries are preserved unchanged.
- Regression coverage: added a batch-response test for an unknown id alongside a matched dumpprivkey response.
- Verification: `./node_modules/.bin/mocha --timeout 20000 --recursive --require should test/logging/request-obfuscator_test.js` passed with 23 tests; `npm run lint` passed; `git diff --check HEAD~1..HEAD` is clean.

Payment details can be provided if accepted or requested.

**Comment by manuelsampedro1** [comment]
Submitted another focused Bitgesell improvement PR under this bounty program:

- PR: https://github.com/BitgesellOfficial/php-bitgesellcore-rpc/pull/3
- Scope: fixes Composer/PSR-4 namespace drift in php-bitgesellcore-rpc by aligning the package autoload roots, Config namespace, response handler class, and response test imports with the Bitgesell namespace used by the source tree.
- Validation: parsed composer.json with Node; checked for stale Denpa namespace / BitgeselldResponse references with `rg`; `git diff --check`. PHP/Composer tests were not run locally because this environment does not have PHP or Composer installed.

**Comment by manuelsampedro1** [comment]
Submitted another focused Bitgesell improvement PR under this bounty program:

- PR: https://github.com/BitgesellOfficial/bitgesell-bip39/pull/3
- Scope: aligns bitgesell-bip39 package metadata with the forked package by correcting the package-lock root name/version, updating the repository URL to BitgesellOfficial, and changing the package description to Bitgesell BIP39.
- Validation: parsed package.json and package-lock.json with Node; checked stale package metadata references with `rg`; `git diff --check`. npm scripts were not run locally because npm is unavailable in this environment.

**Comment by manuelsampedro1** [comment]
Submitted another focused Bitgesell improvement PR under this bounty program:

- PR: https://github.com/BitgesellOfficial/bgld-docker/pull/3
- Scope: fixes bgld-docker startup reliability by making the root privilege-drop guard match the actual `bgl_oneshot` command, and removes stray whitespace after the Dockerfile `ca-certificates` line-continuation.
- Validation: `/bin/sh -n docker-entrypoint.sh`; `git diff --check`; checked stale patterns with `rg`; smoke-tested the root path with stubbed `id`, `chown`, and `gosu` commands.

**Comment by newmattock** [comment]
Submitted another focused Bitgesell improvement PR under this bounty program:

- PR: https://github.com/BitgesellOfficial/wbgl-bridge-sdk/pull/4
- Scope: fixes `Transaction.makeTxObject()` so repeated BGL bridge swap transaction builds return a fresh Bitgesell transaction object instead of returning `undefined` after the first SDK initialization.
- Regression coverage: adds a focused Jest test proving repeated factory calls return distinct transaction objects.
- Verification: `pnpm exec jest src/__tests__/Transaction.test.ts --runInBand`; `pnpm exec tsc --noEmit --skipLibCheck --module commonjs --target es2019 --esModuleInterop --types jest src/bridge/Transaction.ts src/__tests__/Transaction.test.ts`; `pnpm exec prettier --check src/bridge/Transaction.ts src/__tests__/Transaction.test.ts`; `git diff --check`.

Notes: full project `tsc -p tsconfig.json --skipLibCheck` currently fails on the upstream-visible `@types/node` dependency gap, and focused ESLint did not complete locally after roughly 45 seconds, so the submitted proof uses focused test/typecheck/format/diff gates.

**Comment by newmattock** [comment]
Submitted another focused Bitgesell improvement PR under this bounty program:

- PR: https://github.com/BitgesellOfficial/wbgl-bridge-sdk/pull/5
- Scope: makes `WBGL` honor `config.bridgeEndpoint`, matching the rest of the SDK instead of always submitting WBGL swaps to the production default endpoint.
- Regression coverage: adds a mocked WBGL swap test proving the submit request goes to the configured bridge endpoint while stubbing the token-send path.
- Verification: `pnpm exec jest src/__tests__/WBGLEndpoint.test.ts --runInBand`; `pnpm exec tsc --noEmit --skipLibCheck --module commonjs --target es2019 --esModuleInterop --types jest src/bridge/WBGL.ts src/__tests__/WBGLEndpoint.test.ts`; `pnpm exec prettier --check src/__tests__/WBGLEndpoint.test.ts`; `git diff --check`.

**Comment by newmattock** [comment]
Submitted another focused Bitgesell improvement PR under this bounty program:

- PR: https://github.com/BitgesellOfficial/bitgesell-blockchain-sdk/pull/5
- Scope: updates the source SDK package metadata and README docs link from the old personal namespace to the official BitgesellOfficial repositories.
- Validation: stale-link scan for `naftalimurgor/bitgesell-blockchain-sdk` and `naftalimurgor.github.io/bitgesell-blockchain-sdk` is clean; package JSON and lockfile parse; both official repositories resolve through `gh repo view`; `git diff --check` passes.

**Comment by manuelsampedro1** [comment]
Submitted another focused Bitgesell improvement PR under this bounty program:

- PR: https://github.com/BitgesellOfficial/bglutil/pull/5
- Scope: refreshes the bglutil README so the package title, badges, description, install command, and development instructions point to the Bitgesell module instead of stale btcsuite/btcutil and Conformal release-verification references.
- Validation: checked stale README references with `rg`; `git diff --check`. Go tests were not run locally because the Go toolchain is not installed in this environment.

**Comment by manuelsampedro1** [comment]
Submitted another focused Bitgesell improvement PR under this bounty program:

- PR: https://github.com/BitgesellOfficial/go-bgld/pull/4
- Scope: refreshes the go-bgld README by replacing stale upstream go-bitcoind/Godoc references, modernizing the test command to `go test ./...`, and making the usage example valid Go with concrete Bitgesell defaults.
- Validation: checked stale README references with `rg`; `git diff --check`. Go tests were not run locally because the Go toolchain is not installed in this environment.

**Comment by manuelsampedro1** [comment]
Submitted another PR for the Bitgesell general PR bounty:

https://github.com/BitgesellOfficial/tbgl-v2/pull/1

This one fixes the `tbgl-v2` README workflow badges so they report the Bitgesell repository/`bgl` branch status rather than the upstream `keep-network/tbtc-v2` repo, and corrects the broken Yearn badge query parameter (`vent=push` -> `event=push`).

Validation run:
- `rg -n "keep-network/tbtc-v2/actions|[?&]vent=|branch=main" README.adoc && exit 1 || true`
- `git diff --check`

**Comment by g8rr5dg2p7-svg** [comment]
Submitted a focused Bitgesell improvement PR under this bounty program:

- PR: https://github.com/BitgesellOfficial/bitgesell/pull/279
- Scope: updates `doc/cjdns.md` so the CJDNS guide uses BGL Core / Bitgesell wording, `BGL-cli` commands, and the local Tor guide link instead of inherited Bitcoin Core / `bitcoin-cli` references.
- Validation: `git diff --check HEAD~1..HEAD`, stale Bitcoin wording scan, and `python3 test/lint/lint-tests.py` all passed.

Payment details can be provided if accepted or requested.

**Comment by tolga-tom-nook** [comment]
Submitted another focused JS RPC client fix for Bitgesell bounty consideration:

- PR: https://github.com/BitgesellOfficial/js-bitgesellcore-rpc/pull/7
- Scope: fixes `Client#command()` multiwallet routing for mixed-case RPC method names. Direct calls like `command('getBalance')` were sent as lowercase `getbalance`, but the wallet-support lookup used the original mixed-case key and missed multiwallet support, so requests posted to `/` instead of `/wallet/<wallet>`.
- Regression coverage: single and batch mixed-case wallet commands now assert the selected wallet URI is used.
- RED proof on `origin/main`: new tests failed with `AssertionError: expected '/' to be '/wallet/wallet.dat'`.
- GREEN proof after fix: `./node_modules/.bin/mocha test/command_wallet_routing_test.js --timeout 20000 --require should` passed (`2 passing`); `./node_modules/.bin/eslint src/index.js test/command_wallet_routing_test.js` passed; `git diff --check` clean.
- Payout address if accepted: EVM/USDT-compatible `0x4a76c7E64C08cF29B59eFC640b4ada97A270d428`.

Assisted by Hermes Agent.

**Comment by jtc268** [comment]
Submitted PR #289 for this bounty: https://github.com/BitgesellOfficial/bitgesell/pull/289

It aligns the remaining network docs in touched files with BGL naming: BGL Core, BGL-cli, BGL.conf, and BGL node terminology. Validation: git diff --check plus grep for the stale Bitcoin command/config/product strings in the touched docs.

**Comment by tolga-tom-nook** [comment]
Submitted a new Bitgesell API contribution for bounty review:

- PR: https://github.com/BitgesellOfficial/bglapiserver/pull/8
- Fix: impossible block-header `count` validation (`count < 1 and count > 2000`) changed to a working lower/upper bound guard, with regression coverage for `0` and `5000` count values.
- Requested liquid payout rail if accepted: USDT / USDC-compatible EVM address `0x4a76c7E64C08cF29B59eFC640b4ada97A270d428`

Thank you — happy to adjust if maintainers prefer a different count handling behavior.

**Comment by JONASXZB** [comment]
Submitted PR #330 for the PR bounty hunt: https://github.com/BitgesellOfficial/bitgesell/pull/330

Scope: small test/docs consistency cleanup replacing stale `test_bitcoin` references with the actual Bitgesell `test_BGL` unit-test binary name, fixing a malformed `test_ --run_test` example, and aligning the LCOV docs with `test_BGL.coverage`.

Validation: stale-reference grep is clean for the touched scope, coverage docs match `Makefile.am`, and `git diff --check` passes.

If this qualifies, public payout addresses are:
- USDT TRC20: `TU3WsDnY9RbwGFEHx6WoF1C7J592SGrS8F`
- USDT/ERC20-compatible EVM: `0xc580927355132643ad82b4c12ba94a7a93e63ba2`

**Comment by Chillzyg** [comment]
Submitted a focused bridge service reliability PR for Bitgesell bounty consideration:

- PR: https://github.com/BitgesellOfficial/wbgl-bridge/pull/40
- Scope: fixes async nonce initialization in the WBGL bridge service so startup does not store unresolved `getTransactionCount()` promises as local nonces, reserves the current nonce before incrementing, and accepts `ETH_NONCE` while preserving the existing `ETH__NONCE` fallback.
- Validation: `npx mocha src/tests/nonce.js --timeout 10000`, targeted Prettier check, targeted `node --check`, targeted ESLint with ES-module parser options, and `git diff --check` all passed.

Payment details can be provided if accepted or requested.

**Comment by tolga-tom-nook** [comment]
Submitted another focused Bitgesell bounty PR:

- PR: https://github.com/BitgesellOfficial/go-bgld/pull/5
- Repo: `go-bgld`
- Fix: `GetBlockTemplate` now returns the `getblocktemplate` RPC result instead of always returning an empty string on success.
- Validation: added regression test; `go test -run TestGetBlockTemplateReturnsRPCResult -count=1 -v` and `go test ./...` pass locally.

Public payout address for USDT/USDC if accepted: `0x4a76c7E64C08cF29B59eFC640b4ada97A270d428` (EVM/BSC).

**Comment by tolga-tom-nook** [comment]
Submitted another focused Bitgesell API contribution for bounty review:

- PR: https://github.com/BitgesellOfficial/bglapiserver/pull/9
- Fix: `GET /blocks/last/{n}` and `GET /blocks/data/last/{n}` now reject zero, negative, and non-numeric counts with `PARAMETER_ERROR` instead of falling through to generic internal-error behavior or the model layer.
- Validation: `python3 -m py_compile api/views/blocks.py`; synthetic handler check verified `0`, `-1`, and `abc` return HTTP 400 for both affected handlers without calling model functions.

**Comment by tolga-tom-nook** [comment]
Submitted another focused Bitgesell bounty PR:

- PR: https://github.com/BitgesellOfficial/keep-core/pull/1
- Repo: `keep-core`
- Fix: `TransactionBuilder.getScript` now validates the referenced output index and returns a descriptive error instead of panicking when a UTXO points at a missing output.
- Validation: added regression test; `go test ./pkg/bitcoin -run 'TestTransactionBuilder_AddPublicKeyHashInput_OutOfRangeOutputIndex|TestTransactionBuilder_AddPublicKeyHashInput|TestTransactionBuilder_AddScriptHashInput' -count=1`, `go test ./pkg/bitcoin -count=1`, and `git diff --check` pass locally.

Public payout address for USDT/USDC if accepted: `0x4a76c7E64C08cF29B59eFC640b4ada97A270d428` (EVM/BSC).

**Comment by modelsbridgeaicom-ship-it** [comment]
Submitted a maintenance PR for bounty review:

- PR: https://github.com/BitgesellOfficial/nucypher/pull/1
- Change: align setup.py `python_requires` with the rest of the project metadata (`pyproject.toml` and requirements markers already require Python >=3.9,<4)
- Why it matters: prevents package metadata from advertising unsupported Python 3.7/3.8 installs
- Validation: `python -m py_compile setup.py`; `git diff --check`

Payout details can be provided if maintainers accept the PR and request them.

**Comment by modelsbridgeaicom-ship-it** [comment]
Submitted another focused Bitgesell improvement PR under this bounty program:

- PR: https://github.com/BitgesellOfficial/jsbgl/pull/6
- Scope: validates Shamir split parameters before share generation, rejecting non-integer threshold/total values and requiring both threshold and total to be at least 2.
- Why it matters: `__restore_secret` already requires at least two unique points, but `__split_secret` accepted degenerate 1-of-N requests and invalid totals before this change.
- Regression coverage: added invalid-parameter assertions and updated the embedded-index mnemonic test to expect 1-of-N splits to fail fast.
- Validation: `npm.cmd test` passed with 69 tests; `git diff --check` passed.

Payout details can be provided if maintainers accept the PR and request them.

**Comment by modelsbridgeaicom-ship-it** [comment]
Submitted another focused Bitgesell improvement PR under the bounty program:

- PR: https://github.com/BitgesellOfficial/pybgl/pull/20
- Scope: makes pybgl Bech32/SegWit decoding helpers accept all-uppercase `BGL1...` addresses consistently, matching the lowercase hash/script/type/network/witness-version results.
- Why it matters: Bech32 permits all-uppercase addresses, but the current decode helpers only match lowercase `bgl` / `tbgl` prefixes and can return `None` / `UNKNOWN` on otherwise valid uppercase addresses.
- Validation: `python -m compileall -q pybgl tests.py setup.py setup_tools.py`; focused RED/GREEN regression with local native-extension stubs showed `origin/master` failing uppercase decode paths and this branch passing them; `git diff --check` passed with Windows line-ending warnings only.

Payment details can be provided if maintainers accept the PR and request them.

**Comment by cedar323** [comment]
Submitted a focused js-bitgesellcore-rpc maintenance PR under this bounty program:

- PR: https://github.com/BitgesellOfficial/js-bitgesellcore-rpc/pull/11
- Scope: makes `npm test` cross-platform by replacing the POSIX-only `NODE_ENV=test mocha $npm_package_options_mocha` script with a small Node runner that preserves the existing Mocha options and forwards extra CLI arguments.
- Validation: `npm test -- test\errors\rpc-error_test.js --reporter spec` passes on Windows; `git diff --check` reports no patch whitespace errors beyond Git's local CRLF warning.

Payout details can be provided after acceptance/request.

**Comment by cedar323** [comment]
Submitted another focused js-bitgesellcore-rpc test-fix PR under this bounty program:

- PR: https://github.com/BitgesellOfficial/js-bitgesellcore-rpc/pull/12
- Scope: fixes `test/parser_test.js` after the project config was renamed to `bitgesell` / `bitgesellMultiWallet`, and mocks the remaining parser RPC responses so the parser test file runs without a local BGLd/Docker service.
- Validation: `$env:NODE_ENV='test'; node node_modules\mocha\bin\mocha --timeout 20000 --recursive --require should test\parser_test.js --reporter spec` passes on Windows; `git diff --check` reports no patch whitespace errors beyond Git's local CRLF warning.

Payout details can be provided after acceptance/request.

**Comment by cedar323** [comment]
Submitted another focused Bitgesell test/build fix under this bounty program:

- PR: https://github.com/BitgesellOfficial/bitgesell-bip39/pull/4
- Scope: makes `npm test` work on Windows by replacing the POSIX-only `rm -rf src` clean command with a Node cleanup script and switching the Prettier TypeScript glob from single quotes to double quotes for Windows npm shells.
- Validation: reproduced the original Windows failure (`'rm' is not recognized`), then verified `npm test` passes on Windows end-to-end: build, format check, lint, and all 383 tape assertions. `git diff --check` reports no patch whitespace errors beyond Git's local CRLF warning.

Payout details can be provided after acceptance/request.

**Comment by ManuelSalcedo2770** [comment]
Submitted a focused Bitgesell SDK maintenance PR under this bounty program:

- PR: https://github.com/BitgesellOfficial/bitgesell-blockchain-sdk/pull/8
- Scope: restores normal npm installation by aligning Jest/ts-jest tooling, updates TypeDoc to match the repo's TypeScript range, and removes obsolete TypeDoc CLI flags.
- Verification: npm ci, npm run build, npm test -- --runInBand --coverage=false, npm run doc:json, git diff --check.

Payout details can be provided after maintainer review/approval.

**Comment by ManuelSalcedo2770** [comment]
Submitted another focused Bitgesell SDK maintenance PR under this bounty program:

- PR: https://github.com/BitgesellOfficial/wbgl-bridge-sdk/pull/7
- Scope: restores clean npm installation by adding a root lockfile, aligning the ESLint plugin with ESLint 8, pinning Node types to the repo's TypeScript range, and updating the TypeDoc JSON command flags.
- Verification: npm ci, npm run build, npm run doc:json, git diff --check.

Payout details can be provided after maintainer review/approval.

**Comment by ManuelSalcedo2770** [comment]
Submitted another small Bitgesell wallet documentation/metadata PR under this bounty program:

- PR: https://github.com/BitgesellOfficial/bitgesell-wallet/pull/10
- Scope: points package issue/source metadata and README jsbgl links to the official BitgesellOfficial repositories.
- Verification: npm test, npm run build, git diff --check, and GitHub API checks for the official repositories.

This is intentionally scoped as a smaller cleanup. Payout details can be provided after maintainer review/approval.

**Comment by cedar323** [comment]
Submitted another focused maintenance PR for the bounty:

- PR: https://github.com/BitgesellOfficial/assets/pull/1
- Scope: updates `sharp` from `^0.25.4` to `^0.32.6` so the asset test suite can load the native image dependency on current Node/Windows runtimes.
- Validation on Windows / Node v22.18.0:
  - `npm ci`
  - `npm test` (3 suites, 20 tests passed)
  - `npm run lint`

No payout or private details included here.

**Comment by Cycle1337** [comment]
Bounty submission for the PR bounty hunt:

https://github.com/BitgesellOfficial/bitgesell/pull/358

Scope: fixes the Debian packaging metadata behind #357 by removing obsolete/development packages from runtime dependencies (`perl-modules-5.34`, `libsqlite3-dev`, `libqrencode-dev`) and moving needed dev libraries to `Build-Depends`.

Validation:
- Parsed `debian.minimal/control` and `debian.qt/control` to confirm binary `Depends` no longer include the obsolete/dev packages.
- Parsed source `Build-Depends` to confirm build-time dependencies remain available.
- `git diff --check` passed.

Payment details can be provided after maintainer review/approval.

**Comment by nexicturbo** [comment]
Submitted two focused Bitgesell test/lint fixes for bounty review:

- PR #359: https://github.com/BitgesellOfficial/bitgesell/pull/359
  - Scope: makes `test/lint/lint-includes.py` pass by removing duplicate `src/span.h` includes, removing obsolete `boost/process.hpp` from the expected Boost include list, and converting lint-reported include syntax.
  - Validation: `python test\lint\lint-includes.py`; `git diff --check`.

- PR #360: https://github.com/BitgesellOfficial/bitgesell/pull/360
  - Scope: makes `test/lint/lint-format-strings.py` runnable and green by invoking the helper through the active Python interpreter, updating the Bitgesell `BGL-tidy` exclusion, and excluding the generated secp precompute false-positive source.
  - Validation: `python test\lint\lint-format-strings.py`; `python -m py_compile test\lint\lint-format-strings.py test\lint\run-lint-format-strings.py`; `git diff --check`.

Both PRs are open, non-draft, and mergeable. Payment details can be provided after maintainer approval.

**Comment by 220nightmore-spec** [comment]
Submitted PR #361 for this bounty program: https://github.com/BitgesellOfficial/bitgesell/pull/361

Scope: documentation/template cleanup only. It updates the GitHub issue templates so new bug, GUI, and good-first-issue reports point to Bitgesell/BGL Core resources instead of inherited Bitcoin Core locations.

Validation:
- `git diff --check`
- Checked the edited templates for required issue-template keys and tab-free indentation.
- Confirmed the edited templates no longer contain Bitcoin Core/bitcoin-core issue tracker links.

Preferred payout if approved: USDT ERC20 or BEP20/BSC to `0x185e250a54ced0d999cbc3280ca8481b35aa2ce0`

**Comment by 220nightmore-spec** [comment]
Submitted another focused Bitgesell documentation maintenance PR for bounty review: https://github.com/BitgesellOfficial/bitgesell-blockchain-sdk-docs/pull/2

Scope: updates the generated itgesell-blockchain-sdk-docs output so TypeDoc source links point to the official Bitgesell SDK repository instead of the old personal source repository, aligns the install command with the published package name, and fixes small README/example typos.

Validation:
- git diff --check
- confirmed no remaining 
aftalimurgor/bitgesell-blockchain-sdk, gl-blockchain-sdk, Bockchain, or To query transacton references in the docs output
- checked representative linked source paths exist in the official SDK checkout

Preferred payout if approved: USDT on Ethereum/ERC20 or BSC/BEP20 to x185e250a54ced0d999cbc3280ca8481b35aa2ce0.

**Comment by 220nightmore-spec** [comment]
Submitted another small Bitgesell documentation maintenance PR for bounty review: https://github.com/BitgesellOfficial/keep-core-bgl/pull/3

Scope: updates keep-core-bgl README metadata so the title, GitHub Actions badges, and feedback issue link point at the BitgesellOfficial fork instead of upstream keep-network/keep-core.

Validation:
- git diff --check
- verified the referenced workflow files exist locally
- confirmed no README badge or issue-feedback links still point to upstream keep-network/keep-core

Preferred payout if approved: USDT on Ethereum/ERC20 or BSC/BEP20 to 0x185e250a54ced0d999cbc3280ca8481b35aa2ce0.

**Comment by 220nightmore-spec** [comment]
Correction on my previous keep-core-bgl submission: I closed https://github.com/BitgesellOfficial/keep-core-bgl/pull/3 because GitHub showed an unexpectedly broad fork/base diff instead of the intended one-file README metadata update. Please do not consider that PR for bounty review. The valid current submissions from me remain PR #361 in bitgesell and PR #2 in bitgesell-blockchain-sdk-docs.

**Comment by 220nightmore-spec** [comment]
Correction for my previous Bitgesell SDK docs PR comment: the valid payout address is:

USDT ERC20 or BEP20/BSC: `0x185e250a54ced0d999cbc3280ca8481b35aa2ce0`

Valid current submissions from me for this bounty review:
- https://github.com/BitgesellOfficial/bitgesell/pull/361
- https://github.com/BitgesellOfficial/bitgesell-blockchain-sdk-docs/pull/2

The closed keep-core-bgl PR should not be considered.

**Comment by 220nightmore-spec** [comment]
Additional valid PRs for the bounty hunt:

- https://github.com/BitgesellOfficial/keep-core-bgl/pull/4
- https://github.com/BitgesellOfficial/tbgl-v2/pull/2
- https://github.com/BitgesellOfficial/bgld-docker/pull/5
- https://github.com/BitgesellOfficial/bitgesell-arm/pull/3

keep-core-bgl summary: updates README and CONTRIBUTING links so the title, CI badges, issue link, contribution fork URL, development-doc URL, Actions URL, and target branch guidance point at the Bitgesell fork and the `bgl` branch instead of upstream `keep-network/keep-core` / `main`.

keep-core-bgl validation:
- PR readback: open, non-draft, mergeable, `changed_files=2`, `additions=15`, `deletions=11`, base `bgl`.
- Updated GitHub workflow, repository, branch, and issues links return HTTP 200.
- `git diff --check` passes.

tbgl-v2 summary: updates CONTRIBUTING fork URL, target branch guidance, and GitHub Actions links to point at `BitgesellOfficial/tbgl-v2` and the `bgl` branch instead of upstream `keep-network/tbtc-v2` / `main`.

tbgl-v2 validation:
- PR readback: open, non-draft, mergeable, `changed_files=1`, `additions=4`, `deletions=4`, base `bgl`.
- Updated repository and Actions links return HTTP 200.
- `git diff --check` passes.

bgld-docker summary: updates CONTRIBUTING links from the old Travis/upstream docker-bgld references to this repository's GitHub Actions page and test directory.

bgld-docker validation:
- PR readback: open, non-draft, mergeable, `changed_files=1`, `additions=2`, `deletions=2`, base `main`.
- Updated GitHub Actions and test-directory links return HTTP 200.
- `git diff --check` passes.
- Existing open PR #1 only touches README/test README and does not cover `CONTRIBUTING.md`.

bitgesell-arm summary: updates README image/bootstrap/docs links and CONTRIBUTING CI/test links to point at the official `BitgesellOfficial/bitgesell-arm` repository.

bitgesell-arm validation:
- PR readback: open, non-draft, mergeable, `changed_files=2`, `additions=5`, `deletions=5`, base `main`.
- Updated image, bootstrap, docs, Actions, and test-directory links return HTTP 200.
- `git diff --check` passes.
- Existing open PR #1 only changes entrypoint logic/tests and does not cover these docs links.

USDT ERC20/BEP20: `0x185e250a54ced0d999cbc3280ca8481b35aa2ce0`

**Comment by wangedmund77-cmyk** [comment]
Submitted a focused Bitgesell devtools fix for bounty review:

- PR: https://github.com/BitgesellOfficial/bitgesell/pull/363
- Scope: fixes `contrib/devtools/gen-BGL-conf.sh` so it writes `share/examples/BGL.conf`, emits BGL-specific generated headers, and parses options from the configured `BGLD` binary instead of the undefined `${BITCOIND}` variable.
- Docs: aligns `contrib/devtools/README.md` and `share/examples/BGL.conf` placeholder text with `gen-BGL-conf.sh` / `BGL.conf`.
- Validation: `bash -n contrib/devtools/gen-BGL-conf.sh`; `git diff --check`; smoke test with a temporary mock `BGLd` confirming generated BGL headers, parsed option lines, and no `bitcoin.conf` output.

Payout details can be provided after maintainer approval.

**Comment by 220nightmore-spec** [comment]
For the active PR bounty hunt, can a maintainer confirm whether an approved contribution may be paid as USDT on Ethereum/ERC20 or BSC/BEP20? My previously published payout preference is the EVM address already listed in this thread; I cannot receive TRC20. This is only a payout-network confirmation request, not a payment claim.

**Comment by 220nightmore-spec** [comment]
Submitted a focused documentation/package-metadata PR for bounty review: https://github.com/BitgesellOfficial/pybgl/pull/21

Scope: points pybgl README, package metadata, Sphinx source links, clone commands, and issue tracker links from the former bitaps-com repository to BitgesellOfficial/pybgl.

Validation: git diff --check; python -m compileall -q setup.py.

Preferred payout remains USDT ERC20 or BEP20/BSC to the EVM address already published in this thread. No payment is claimed by this submission.
